consensus/ibft/ibft.go

```diff
@@ -295,7 +294,7 @@ func (i *backendIBFT) startConsensus() {
 		}
 
 		// Update the No.of validator metric
-		i.metrics.Validators.Set(float64(i.currentValidators.Len()))
+		metrics.SetGauge([]string{"validators"}, float32(i.currentValidators.Len()))
```

Comment:
Just curious, is it a thread-safe operation? Won't we get data races in tests?

Reply:
No data races, it is thread-safe.

Reply:
I believe we don't have tests with race at the moment. Could we check?

Reply:
I tested with the `txpool`, there are data races there but none related to the `metrics`. This library is used across many OSS projects (including Hashicorp tools) without explicit locks, so it is safely handled internally.